src/php/lib/Grpc/BaseStub.php

```diff
@@ -203,10 +213,134 @@ private function _validate_and_normalize_metadata($metadata)
 
         return $metadata_copy;
     }
+  
+    private function GrpcUnaryUnary($channel){
+        $this->current_channel = $channel;
```

Comment:
I have a feeling this should be `$current_channel` local variable that is then captured by the closure function returned by this method, not mutable state attached to the stub.